x-pack/test/alerting_api_integration/spaces_only/tests/alerting/alerts_base.ts

```diff
@@ -135,6 +136,141 @@ instanceStateValue: true
+              },
+              {
+                group: ResolvedActionGroup.id,
+                id: indexRecordActionId,
+                params: {
+                  index: ES_TEST_INDEX_NAME,
+                  reference,
+                  message: 'Resolved message',
+                },
+              },
+            ],
+          })
+        );
+
+      expect(createdAlert.status).to.eql(200);
+      const alertId = createdAlert.body.id;
+      objectRemover.add(space.id, alertId, 'alert', 'alerts');
+
+      const actionTestRecord = (
+        await esTestIndexTool.waitForDocs('action:test.index-record', reference)
+      )[0];
+
+      expect(actionTestRecord._source).to.eql({
+        config: {
+          unencrypted: `This value shouldn't get encrypted`,
+        },
+        secrets: {
+          encrypted: 'This value should be encrypted',
+        },
+        params: {
+          index: ES_TEST_INDEX_NAME,
+          reference,
+          message: 'Resolved message',
+        },
+        reference,
+        source: 'action:test.index-record',
+      });
+    });
+
+    it('should not fire actions when an alert instance is resolved, but alert is muted', async () => {
+      const reference = alertUtils.generateReference();
+
+      const { body: createdAction } = await supertestWithoutAuth
+        .post(`${getUrlPrefix(space.id)}/api/actions/action`)
+        .set('kbn-xsrf', 'foo')
+        .send({
+          name: 'MY action',
+          actionTypeId: 'test.noop',
+          config: {},
+          secrets: {},
+        })
+        .expect(200);
+
+      // pattern of when the alert should fire.
+      const pattern = {
+        instance: [false, true, true],
+      };
+
+      const createdAlert = await supertestWithoutAuth
+        .post(`${getUrlPrefix(space.id)}/api/alerts/alert`)
+        .set('kbn-xsrf', 'foo')
+        .send(
+          getTestAlertData({
+            alertTypeId: 'test.patternFiring',
+            schedule: { interval: '1s' },
+            throttle: null,
+            params: {
+              pattern,
+            },
+            actions: [
+              {
+                id: createdAction.id,
+                group: 'default',
+                params: {},
+              },
+              {
+                group: ResolvedActionGroup.id,
+                id: indexRecordActionId,
+                params: {
+                  index: ES_TEST_INDEX_NAME,
+                  reference,
+                  message: 'Resolved message',
+                },
+              },
+            ],
+          })
+        );
+
+      await alertUtils.muteAll(createdAlert.id);
+
+      expect(createdAlert.status).to.eql(200);
+      const alertId = createdAlert.body.id;
+      objectRemover.add(space.id, alertId, 'alert', 'alerts');
+
+      const actionTestRecord = (
+        await esTestIndexTool.waitForDocs('action:test.index-record', reference, 0)
+      )[0];
+      expect(actionTestRecord).to.eql(0);
```

Comment:
To reduce potential of flakiness, I'm wondering if this test should go as follows:
- Create a disabled alert
- Call the mute all API
- Enable the alert to start running the scenario
- Wait until the alert executed three times (either by implementing something like [this](https://github.com/elastic/kibana/blob/master/x-pack/test/alerting_api_integration/common/fixtures/plugins/alerts/server/alert_types.ts#L79-L89) or by asserting records in the event log)
- `await taskManagerUtils.waitForActionTaskParamsToBeCleanedUp(testStart);`
- `await esTestIndexTool.search('action:test.index-record', reference)` and ensure 0 documents are returned

I think the waiting for the alert to run three times / asserting on event log data will confirm the second run did not schedule actions for the alert instance. I can see a scenario where the above would complete the test before the alert finishes running.